packages/sanity/src/credentials.ts

```diff
@@ -0,0 +1,79 @@
+import Providers, { CredentialsProvider } from "next-auth/providers"
+import type { SanityClient } from "@sanity/client"
+import { getUserByEmailQuery } from "./queries"
+import argon2 from "argon2"
+import type { IncomingMessage, ServerResponse } from "http"
+
+interface Options {
+  client: SanityClient
+}
+
+type CredentialsConfig = ReturnType<CredentialsProvider>
+
+export interface Handler {
+  req: IncomingMessage & { body: any }
+  res: ServerResponse & {
+    json: (body: any) => void
+  }
+  client: SanityClient
+}
+
+export const signUpHandler = async ({ req, client, res }: Handler) => {
+  const { email, password, name, image } = req.body
+
+  const user = await client.fetch(getUserByEmailQuery, {
+    email,
+  })
+
+  if (user) {
+    res.json({ error: "User already exist" })
+    return
+  }
+
+  const newUser = await client.create({
+    _type: "user",
+    email,
+    password: await argon2.hash(password),
+    name,
+    image,
+  })
+
+  res.json({
+    email: newUser.email,
+    name: newUser.name,
+    image: newUser.image,
+  })
+}
```

Comment:
```suggestion
export const signUpHandler = (req, res) => async ({ client }: Options) => {
  const { email, password, name, image } = req.body

  const user = await client.fetch(getUserByEmailQuery, {
    email,
  })

  if (user) {
    return res.json({ error: "User already exist" })
  }

  const newUser = await client.create({
    _type: "user",
    email,
    password: await argon2.hash(password),
    name,
    image,
  })

  res.json({
    email: newUser.email,
    name: newUser.name,
    image: newUser.image,
  })
}
```

A suggestion to simplify this logic, so the user can use this method as follows:

```js
import { signUpHandler } from "@next-auth/sanity-adapter"
import { client } from "your/sanity/client"

export default signUpHandler({client})
```

I suggested the corresponding example changes as well.